### ISSUE TITLE
TypeError: 'method' object is not iterable'

### ISSUE THREAD

**Issue by VirajOke** [jerryjliu/llama_index#3931]
**Code snippet** 
```
prompt = "can you summarize the documents?"
query_engine = index.as_query_engine()
response = query_engine.query(prompt)
display(Markdown(f"<b>{response}</b>"))
```
`TypeError: 'method' object is not iterable`

My code was working fine with the previous querying structure. 
i.e. index.query(). 

The new method raises the `method' object is not iterable` error.

**Comment by logan-markewich** [comment]
Do you have the full traceback?

**Comment by Fred-Nuno** [comment]
Having the same issue.

Traceback too long though.

```python

Cell In[80], line 3
      1 query_engine = index.as_query_engine()
      2 # response = query_engine.query("What is the patient name?")
----> 3 response = query_engine.query(str("What is the patient name?"))
      4 print(response)

Lib/site-packages/llama_index/indices/query/base.py:23), in BaseQueryEngine.query(self, str_or_query_bundle)
     21 if isinstance(str_or_query_bundle, str):
     22     str_or_query_bundle = QueryBundle(str_or_query_bundle)
---> 23 response = self._query(str_or_query_bundle)
     24 return response

Lib/site-packages/llama_index/query_engine/retriever_query_engine.py:145), in RetrieverQueryEngine._query(self, query_bundle)
    140 nodes = self._retriever.retrieve(query_bundle)
    141 self.callback_manager.on_event_end(
    142     CBEventType.RETRIEVE, payload={"nodes": nodes}, event_id=retrieve_id
    143 )
--> 145 response = self._response_synthesizer.synthesize(
    146     query_bundle=query_bundle,
    147     nodes=nodes,
    148 )
    150 self.callback_manager.on_event_end(CBEventType.QUERY, event_id=query_id)
    151 return response

Lib/site-packages/llama_index/indices/query/response_synthesis.py:178), in ResponseSynthesizer.synthesize(self, query_bundle, nodes, additional_source_nodes)
    176 if self._response_mode != ResponseMode.NO_TEXT:
    177     assert self._response_builder is not None
--> 178     response_str = self._response_builder.get_response(
    179         query_str=query_bundle.query_str,
    180         text_chunks=text_chunks,
    181         **self._response_kwargs,
    182     )
    183 else:
    184     response_str = None

Lib/site-packages/llama_index/indices/response/compact_and_refine.py:50), in CompactAndRefine.get_response(self, query_str, text_chunks, prev_response, **response_kwargs)
     48 max_prompt = get_biggest_prompt([text_qa_template, refine_template])
     49 new_texts = self._service_context.prompt_helper.repack(max_prompt, text_chunks)
---> 50 response = super().get_response(
     51     query_str=query_str, text_chunks=new_texts, prev_response=prev_response
     52 )
     53 return response

Lib/site-packages/llama_index/token_counter/token_counter.py:78), in llm_token_counter..wrap..wrapped_llm_predict(_self, *args, **kwargs)
     76 def wrapped_llm_predict(_self: Any, *args: Any, **kwargs: Any) -> Any:
     77     with wrapper_logic(_self):
---> 78         f_return_val = f(_self, *args, **kwargs)
     80     return f_return_val

Lib/site-packages/llama_index/indices/response/refine.py:52), in Refine.get_response(self, query_str, text_chunks, prev_response, **response_kwargs)
     48 for text_chunk in text_chunks:
     49     if prev_response_obj is None:
     50         # if this is the first chunk, and text chunk already
     51         # is an answer, then return it
---> 52         response = self._give_response_single(
     53             query_str,
     54             text_chunk,
     55         )
     56     else:
     57         response = self._refine_response_single(
     58             prev_response_obj, query_str, text_chunk
     59         )

Lib/site-packages/llama_index/indices/response/refine.py:86), in Refine._give_response_single(self, query_str, text_chunk, **response_kwargs)
     81 for cur_text_chunk in text_chunks:
     82     if response is None and not self._streaming:
     83         (
     84             response,
     85             formatted_prompt,
---> 86         ) = self._service_context.llm_predictor.predict(
     87             text_qa_template,
     88             context_str=cur_text_chunk,
     89         )
     90         self._log_prompt_and_response(
     91             formatted_prompt, response, log_prefix="Initial"
     92         )
     93     elif response is None and self._streaming:

Lib/site-packages/llama_index/llm_predictor/base.py:227), in LLMPredictor.predict(self, prompt, **prompt_args)
    222 event_id = self.callback_manager.on_event_start(
    223     CBEventType.LLM,
    224     payload=llm_payload,
    225 )
    226 formatted_prompt = prompt.format(llm=self._llm, **prompt_args)
--> 227 llm_prediction = self._predict(prompt, **prompt_args)
    228 logger.debug(llm_prediction)
    230 # We assume that the value of formatted_prompt is exactly the thing
    231 # eventually sent to OpenAI, or whatever LLM downstream

Lib/site-packages/llama_index/llm_predictor/base.py:195), in LLMPredictor._predict(self, prompt, **prompt_args)
    193 full_prompt_args = prompt.get_full_format_args(prompt_args)
    194 if self.retry_on_throttling:
--> 195     llm_prediction = retry_on_exceptions_with_backoff(
    196         lambda: llm_chain.predict(**full_prompt_args),
    197         [
    198             ErrorToRetry(openai.error.RateLimitError),
    199             ErrorToRetry(openai.error.ServiceUnavailableError),
    200             ErrorToRetry(openai.error.TryAgain),
    201             ErrorToRetry(
    202                 openai.error.APIConnectionError, lambda e: e.should_retry
    203             ),
    204         ],
    205     )
    206 else:
    207     llm_prediction = llm_chain.predict(**full_prompt_args)

Lib/site-packages/llama_index/utils.py:177), in retry_on_exceptions_with_backoff(lambda_fn, errors_to_retry, max_tries, min_backoff_secs, max_backoff_secs)
    175 while True:
    176     try:
--> 177         return lambda_fn()
    178     except exception_class_tuples as e:
    179         traceback.print_exc()

Lib/site-packages/llama_index/llm_predictor/base.py:196), in LLMPredictor._predict..()
    193 full_prompt_args = prompt.get_full_format_args(prompt_args)
    194 if self.retry_on_throttling:
    195     llm_prediction = retry_on_exceptions_with_backoff(
--> 196         lambda: llm_chain.predict(**full_prompt_args),
    197         [
    198             ErrorToRetry(openai.error.RateLimitError),
    199             ErrorToRetry(openai.error.ServiceUnavailableError),
    200             ErrorToRetry(openai.error.TryAgain),
    201             ErrorToRetry(
    202                 openai.error.APIConnectionError, lambda e: e.should_retry
    203             ),
    204         ],
    205     )
    206 else:
    207     llm_prediction = llm_chain.predict(**full_prompt_args)

/Lib/site-packages/langchain/chains/llm.py:213), in LLMChain.predict(self, callbacks, **kwargs)
    198 def predict(self, callbacks: Callbacks = None, **kwargs: Any) -> str:
    199     """Format prompt with kwargs and pass to LLM.
    200 
    201     Args:
   (...)
    211             completion = llm.predict(adjective="funny")
    212     """
--> 213     return self(kwargs, callbacks=callbacks)[self.output_key]

Lib/site-packages/langchain/chains/base.py:140), in Chain.__call__(self, inputs, return_only_outputs, callbacks)
    138 except (KeyboardInterrupt, Exception) as e:
    139     run_manager.on_chain_error(e)
--> 140     raise e
    141 run_manager.on_chain_end(outputs)
    142 return self.prep_outputs(inputs, outputs, return_only_outputs)

Lib/site-packages/langchain/chains/base.py:134), in Chain.__call__(self, inputs, return_only_outputs, callbacks)
    128 run_manager = callback_manager.on_chain_start(
    129     {"name": self.__class__.__name__},
    130     inputs,
    131 )
    132 try:
    133     outputs = (
--> 134         self._call(inputs, run_manager=run_manager)
    135         if new_arg_supported
    136         else self._call(inputs)
    137     )
    138 except (KeyboardInterrupt, Exception) as e:
    139     run_manager.on_chain_error(e)

Lib/site-packages/langchain/chains/llm.py:69), in LLMChain._call(self, inputs, run_manager)
     64 def _call(
     65     self,
     66     inputs: Dict[str, Any],
     67     run_manager: Optional[CallbackManagerForChainRun] = None,
     68 ) -> Dict[str, str]:
---> 69     response = self.generate([inputs], run_manager=run_manager)
     70     return self.create_outputs(response)[0]

Lib/site-packages/langchain/chains/llm.py:79), in LLMChain.generate(self, input_list, run_manager)
     77 """Generate LLM result from inputs."""
     78 prompts, stop = self.prep_prompts(input_list, run_manager=run_manager)
---> 79 return self.llm.generate_prompt(
     80     prompts, stop, callbacks=run_manager.get_child() if run_manager else None
     81 )

/Lib/site-packages/langchain/llms/base.py:134), in BaseLLM.generate_prompt(self, prompts, stop, callbacks)
    127 def generate_prompt(
    128     self,
    129     prompts: List[PromptValue],
    130     stop: Optional[List[str]] = None,
    131     callbacks: Callbacks = None,
    132 ) -> LLMResult:
    133     prompt_strings = [p.to_string() for p in prompts]
--> 134     return self.generate(prompt_strings, stop=stop, callbacks=callbacks)

Lib/site-packages/langchain/llms/base.py:159), in BaseLLM.generate(self, prompts, stop, callbacks)
    154 if not isinstance(prompts, list):
    155     raise ValueError(
    156         "Argument 'prompts' is expected to be of type List[str], received"
    157         f" argument of type {type(prompts)}."
    158     )
--> 159 params = self.dict()
    160 params["stop"] = stop
    161 (
    162     existing_prompts,
    163     llm_string,
    164     missing_prompt_idxs,
    165     missing_prompts,
    166 ) = get_prompts(params, prompts)

Lib/site-packages/langchain/llms/base.py:362), in BaseLLM.dict(self, **kwargs)
    360 def dict(self, **kwargs: Any) -> Dict:
    361     """Return a dictionary of the LLM."""
--> 362     starter_dict = dict(self._identifying_params)
    363     starter_dict["_type"] = self._llm_type
    364     return starter_dict

TypeError: 'method' object is not iterable

```

**Comment by logan-markewich** [comment]
@Fred-Nuno did you define a custom LLM? Please make sure you use the property decorator for the identifying_params attribute.

The correct example is here
https://gpt-index.readthedocs.io/en/latest/how_to/customization/custom_llms.html#example-using-a-custom-llm-model-advanced
